### PR TITLE
add missing methods for `AbstractGenericCharacter`

### PR DESCRIPTION
**Pull request by SoongNoonien**
@fingolfin I think this should be a more complete version of #156.

Closes #156

### CODE REVIEW

File: src/Show.jl [hunk 1]
```diff
@@ -307,9 +307,25 @@ function chardeg(t::Table, char::Int64)
 	if char > chartypes(t)
 		throw(DomainError(char, "Character type is out of range."))
 	end
-	t[char].degree
+	return chardeg(t[char])
 end
 
+@doc raw"""
+    chardeg(char::AbstractGenericCharacter)
```
Comment (fingolfin) [inline src/Show.jl:314]: Can't we just call this "degree" ?

File: src/Show.jl [hunk 2]
```diff
@@ -326,22 +342,36 @@ q - 1
...
-	else
-		o=t.ring(1)
-		result=t[char].sum(o//o)
-		return shrink(result)
-	end
-end
-function nrchars(t::SimpleCharTable, char::Int64)
+function nrchars(t::Table, char::Int64)
 	if char > chartypes(t)
 		throw(DomainError(char, "Character type is out of range."))
```
Comment (fingolfin) [inline src/Show.jl:347]: Is this even needed? Won't `t[char]` throw an error anyway? Perhaps a less nice one -- but if we want a nicer one, I think we should put it into `Base.getindex(ct::CharTable, i::Integer)`, and remove the check here.

**Review by fingolfin — APPROVED** [review]
Yes, much better. Though I left some comments

**Comment by fingolfin** [comment]
I also think we should start moving these methods to better places. Clearly `Show.jl` is not where one would expect methods for characters to be. How about a new file `src/Chars.jl` or `src/Characters.jl` for methods applicable to characters?

And then old `nrchars` etc. which takes a table plus a charcter number should be moved to `src/Deprecations.jl`

File: src/Show.jl (hunk 1 above)
Comment (SoongNoonien) [reply src/Show.jl:314]: Yes, actually I had it this way before but I thought this could be too much change at once. I'll fix it.

**Comment by fingolfin** [comment]
Regarding the "moving things around": we can also just merge this now and you move later. In any case, I'll need the changes in here for the book example.

And I'd like to release 0.4 of this package this week so I can reference it in the book.

File: src/Show.jl (hunk 2 above)
Comment (SoongNoonien) [reply src/Show.jl:347]: No, it is not needed. But we do have this in other functions too. See for example `printinfochar`. Shouldn't we remove it there as well?

Comment (SoongNoonien) [reply src/Show.jl:348]: Ok, I'll do that in a follow up PR where I also remove these checks for the class types.

**Pull request merged**